test/functional/feature_fedpeg.py

```diff
@@ -565,28 +565,44 @@ def run_test(self):
-        assert("valuecommitment" in final_decoded["vout"][0])
-        assert("commitmentnonce" in final_decoded["vout"][0])
-        assert("value" not in final_decoded["vout"][0])
-        assert("asset" not in final_decoded["vout"][0])
-        assert(final_decoded["vout"][0]["commitmentnonce_fully_valid"])
-        assert("assetcommitment" in final_decoded["vout"][1])
-        assert("valuecommitment" in final_decoded["vout"][1])
-        assert("commitmentnonce" in final_decoded["vout"][1])
-        assert("value" not in final_decoded["vout"][1])
-        assert("asset" not in final_decoded["vout"][1])
-        assert(final_decoded["vout"][1]["commitmentnonce_fully_valid"])
-        assert("value" in final_decoded["vout"][2])
-        assert("asset" in final_decoded["vout"][2])
-        blind_pegin_txid = sidechain.sendrawtransaction(pegin_signed["hex"])
-        sidechain.generate(5)
-        assert_equal(sidechain.gettransaction(blind_pegin_txid)["confirmations"], 5)
+        # test with both blindraw and rawblindraw
+        raw_pegin_blinded1 = sidechain.blindrawtransaction(pegin_hex)
+        raw_pegin_blinded2 = sidechain.rawblindrawtransaction(pegin_hex, ["", unspent["amountblinder"]], [10, 15], [unspent["asset"]]*2, ["", unspent["assetblinder"]], "", False)
+        pegin_signed1 = sidechain.signrawtransactionwithwallet(raw_pegin_blinded1)
+        pegin_signed2 = sidechain.signrawtransactionwithwallet(raw_pegin_blinded2)
+        for pegin_signed in [pegin_signed1, pegin_signed2]:
+            final_decoded = sidechain.decoderawtransaction(pegin_signed["hex"])
+            assert(final_decoded["vin"][0]["is_pegin"])
+            assert(not final_decoded["vin"][1]["is_pegin"])
+            assert("assetcommitment" in final_decoded["vout"][0])
+            assert("valuecommitment" in final_decoded["vout"][0])
+            assert("commitmentnonce" in final_decoded["vout"][0])
+            assert("value" not in final_decoded["vout"][0])
+            assert("asset" not in final_decoded["vout"][0])
+            assert(final_decoded["vout"][0]["commitmentnonce_fully_valid"])
+            assert("assetcommitment" in final_decoded["vout"][1])
+            assert("valuecommitment" in final_decoded["vout"][1])
+            assert("commitmentnonce" in final_decoded["vout"][1])
+            assert("value" not in final_decoded["vout"][1])
+            assert("asset" not in final_decoded["vout"][1])
+            assert(final_decoded["vout"][1]["commitmentnonce_fully_valid"])
+            assert("value" in final_decoded["vout"][2])
+            assert("asset" in final_decoded["vout"][2])
+            # check that it is accepted in the mempool
+            accepted = sidechain.testmempoolaccept([pegin_signed["hex"]])[0]
+            if not accepted["allowed"]:
+                print("Blinded pegin not allowed in mempool; reject-reason: %s" % accepted["reject-reason"])
+                raise Exception()
```

Comment:
just stick the message inside the `Exception`

Reply:
haha, nice!